app/controllers/charges_controller.rb

```diff
@@ -0,0 +1,26 @@
+class ChargesController < ApplicationController
+  def create
+    amount = params[:amount].delete('.')
+
+    customer = Stripe::Customer.create({
+      name: params[:name],
+      email: params[:email],
+      source: params[:charge][:token]
+    })
+
+    charge = Stripe::Charge.create({
+      customer: customer.id,
+      amount: amount,
+      description: 'NPO Donation',
+      currency: 'usd',
+      metadata: {
+        npo_ein: params[:npoEin],
+        npo_name: params[:npoName]
+      }
+    })
+
+  rescue Stripe::CardError => e
+    flash[:error] = e.message
+    redirect_to new_charge_path
+  end
+end
```

Comment:
This looks great to me! Is this something that will be called on the FE like `/charges?name=john` instead of through `/graphql`

Reply:
Yeah FE will call /charges but pass data in the body instead of key/value in headers.

Reply:
Badass. I will be doing something similar with the image uploading so glad I'm not alone! lol